api/v1alpha1/nnfdatamovementprofile_webhook_test.go

```diff
@@ -0,0 +1,214 @@
+		otherNamespaceName string
+		otherNamespace     *corev1.Namespace
+
+		pinnedResourceName string
+		nnfProfile         *NnfDataMovementProfile
+		newProfile         *NnfDataMovementProfile
+	)
+
+	BeforeEach(func() {
+		pinnedResourceName = "test-pinned-" + uuid.NewString()[:8]
+
+		nnfProfile = &NnfDataMovementProfile{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-" + uuid.NewString()[:8],
+				Namespace: namespaceName,
+			},
+		}
+
+		newProfile = &NnfDataMovementProfile{}
+	})
+
+	BeforeEach(func() {
+		otherNamespaceName = "other-" + uuid.NewString()[:8]
+
+		otherNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: otherNamespaceName,
+			},
+		}
+		Expect(k8sClient.Create(context.TODO(), otherNamespace)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		Expect(k8sClient.Delete(context.TODO(), otherNamespace)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		if nnfProfile != nil {
+			Expect(k8sClient.Delete(context.TODO(), nnfProfile)).To(Succeed())
+			profExpected := &NnfDataMovementProfile{}
+			Eventually(func() error { // Delete can still return the cached object. Wait until the object is no longer present
+				return k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nnfProfile), profExpected)
+			}).ShouldNot(Succeed())
+		}
+	})
+
+	It("should accept system profiles in the designated namespace", func() {
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+	})
+
+	It("should not accept system profiles that are not in the designated namespace", func() {
+		nnfProfile.ObjectMeta.Namespace = otherNamespaceName
+		err := k8sClient.Create(context.TODO(), nnfProfile)
+		Expect(err.Error()).To(MatchRegexp("webhook .* denied the request: incorrect namespace"))
+		nnfProfile = nil
+	})
+
+	It("should accept default=true", func() {
+		nnfProfile.Data.Default = true
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+		Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nnfProfile), newProfile)).To(Succeed())
+		Expect(newProfile.Data.Default).To(BeTrue())
+	})
+	It("should accept default=false", func() {
+		nnfProfile.Data.Default = false
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+		Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(nnfProfile), newProfile)).To(Succeed())
+		Expect(newProfile.Data.Default).ToNot(BeTrue())
+	})
+
+	It("Should not allow a default resource to be pinned", func() {
+		nnfProfile.Data.Default = true
+		nnfProfile.Data.Pinned = true
+
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).ToNot(Succeed())
+		nnfProfile = nil
+	})
+
+	It("Should not allow modification of Data in a pinned resource", func() {
+		nnfProfile.ObjectMeta.Name = pinnedResourceName
+		nnfProfile.ObjectMeta.Namespace = otherNamespaceName
+		nnfProfile.Data.Pinned = true
+
+		Expect(k8sClient.Create(context.TODO(), nnfProfile)).To(Succeed())
+		Eventually(func() error {
```

Comment:
Can you wrap all the Get()s like this? It's rare that they fail, but every once in a while the timing is just right where a straight Expect() will return an error.

Reply:
Sure - all the pinned/default stuff is ripped off from the storage profile tests.